### ISSUE TITLE
Headless Log Machine Status Updates

### ISSUE THREAD

**Issue by bpm1992** [ElvishArtisan/rivendell#416]
It would be great to get status updates from the Headless Log Machines while they are playing. 

Things that I'd like to see:
1. Current Machine Mode 
2. Log Line
3. Now/Next
4. Current track play point  (what time it is during playback) 
5. VU Meter Data

One way that could work would be to write a JSON object at regular intervals in the /rd-bin directory.   If it were a flat file, it could be easily read and not too expensive compared to hitting a CGI script to poll a database and calculate VU meter levels at 500ms intervals.

This would require changes to rdvairplay, rdadmin, and the database schema to store configuration for this status output.

**Comment by ElvishArtisan** [comment]
I agree that some sort of interface that would allow fully dynamic monitoring and control of log machines would be a Good Thing. Let's put it on the list of enhancements to consider post v3.0.x production release.

**Comment by bpm1992** [comment]
Thanks.  This makes sense for timing.  I could see some sort of a pub/sub interface where the remote system connects through a TCP socket.  That way updates would be pushed one way and commands the other way.   

With the new rdpad daemon, the messages could be JSON formatted.

Maybe an extension of RIPC?

**Comment by k2fc** [comment]
I think I would suggest this be taken a step further.  Now that there is a method to run a log machine headless via a daemon... perhaps this should be the way log machines live.  Other systems do this, where the control room UI is just a UI.  

This allows for several studios (or more than one workstation in a studio) to share one log machine.  NexGen has done it since the beginning, AudioVAULT added it with Flex, and WideOrbit does it starting with v4.0.  

If this is something that's going to be given some thought after the 3.0 release, maybe that can be looked at.  I think it would be a great enhancement!

In my fantasy, the SoundPanel audio and the GPIO remain local to the RDAirPlay UI on the workstation, but you could specify which (local or remote) log machine running in rdvairplayd you are controlling.
